includes/blocks/sophi-page-list-item/edit.js

```diff
@@ -0,0 +1,210 @@
+ * @param {boolean}  props.attributes.postExcept          Whether to display post date.
+ * @param {boolean}  props.attributes.postAuthor          Whether to display featured image.
+ * @param {boolean}  props.attributes.postDate            Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.postDateC           Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.featuredImage       Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.linkToFeaturedImage Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.overrideRule        Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.overridePostID      Whether to add post permalink to featured image.
+ * @param {boolean}  props.attributes.overrideExpiry      Whether to add post permalink to featured image.
+ * @param {string}   props.className                      Class name for the block.
+ * @param {Function} props.setAttributes                  Sets the value for block attributes.
+ *
+ * @returns {Function} Render the edit screen
+ */
+const SiteAutomationItemBlockEdit = ({
+	attributes: {
+		postTitle = '',
+		postLink = '',
+		postExcept = '',
+		postAuthor = '',
+		postDate = '',
+		postDateC = '',
+		featuredImage = '',
+		linkToFeaturedImage = false,
+		overrideRule = '',
+		overridePostID = 0,
+		overrideExpiry = 2,
+	},
+	className,
+	setAttributes,
+}) => {
+	const blockProps = useBlockProps();
+	const [showPopup, setShowPopup] = useState(false);
+	const [selectedPost, setSelectedPost] = useState('');
+	const [message, setMessage] = useState({
+		text: '',
+		color: 'green',
+	});
+
+	const onToggle = () => {
+		setShowPopup((showPopup) => !showPopup);
+	};
+
+	const handlePickChanage = (pickedContent) => {
+		setAttributes({
+			overridePostID: pickedContent.length === 0 ? 0 : pickedContent[0].id,
+		});
+		setSelectedPost(pickedContent);
+	};
+
+	const handleOverride = () => {
+		if (overrideRule === '') {
+			setMessage({
+				text: __('Please select override rule.', 'sophi-wp'),
+				color: 'red',
+			});
+			return;
+		}
+		if (overridePostID === 0 && (overrideRule === 'in' || overrideRule === 'replace')) {
+			setMessage({
+				text: __('Please select the post', 'sophi-wp'),
+				color: 'red',
+			});
+			return;
+		}
+
+		// Close the popup & reset the selected post.
+		setShowPopup(false);
+		setSelectedPost('');
+
+		// Set postUpdated to true so this get caught by the parent.
+		setAttributes({
+			overrideExpiry,
+			postUpdated: true,
+		});
+	};
+
+	const overridePopover = showPopup && (
+		<Popover
+			className="override-map-popover"
+			position="bottom center"
+			focusOnMount="firstElement"
+			key="override-popover"
+			expandOnMobile
+			headerTitle={__('Override', 'Override')}
+			onFocusOutside={() => setShowPopup(false)}
+		>
```

Comment:
I would suggest to also add the `anchorRef` prop here and provide the `ref` of the toolbar button.


```js
import { useRef } from '@wordpress/element';
import { Popover, ToolbarButton } from '@wordpress/components';

const BlockEdit = () => {

    const toolbarButtonRef = useRef();

    return (
        <>
            <ToolbarButton ref={toolbarButtonRef} />
            <Popover anchorRef={toolbarButtonRef} />
        </>
    );
}
```

_Note: This is simplified code that doesn't work. it is only meant to show how the `anchorRef` prop works._

This will fix any positioning issues you are having with the popover and will make sure that it gets attached to the toolbar button :)